### DISCUSSION TITLE
Trouble shooting sql_time_limit_ms

### DISCUSSION THREAD

**Discussion by abekenmore** [simonw/datasette#1731] [Q&A]
Hi all! I set up my first Datasette database, got everything up and running but the data is large (millions of rows). The SQL inquiries timeout after one second, which isn't enough time to process the data. I tried to lengthen the query time limit by running this command in my Mac's terminal: 

`datasette mydatabase.db --setting sql_time_limit_ms 5000`
The only thing I subbed from the example code was my database name and the time limit. It started running, and I got this:

```
INFO:     Started server process [28725]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://127.0.0.1:8001 (Press CTRL+C to quit)
^CINFO:     Shutting down
INFO:     Waiting for application shutdown.
INFO:     Application shutdown complete.
INFO:     Finished server process [28725]
```
I shut it down using CTRL+C after 15 hours of it running without resolving (at ^CINFO), the database still has a 1 second query limit. I tried running the code a few times previously as well. Never got any error codes, but it never resolved either. Apologies if this is the wrong forum, but I'm new to Datasette (and Python - everything I've learned has been to set up this database), so any advice on why the command isn't running would be appreciated.

**Comment by brandonrobertz** [comment]
What page are you getting this timeout on? It sounds like you might be getting hung up on the initial home page, not a user query, which can take a really long time to load with huge DBs. Maybe try using an `--inspect-file` or report back with more info?
